contracts/governance/extensions/GovernorCountingFractional.sol

```diff
@@ -0,0 +1,219 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {Governor} from "../Governor.sol";
+import {GovernorCountingSimple} from "./GovernorCountingSimple.sol";
+import {Math} from "../../utils/math/Math.sol";
+
+/**
+ * @dev Extension of {Governor} for fractional 3 option vote counting.
+ *
+ * Voters can split their voting power amongst 3 options: For, Against and Abstain.
+ * This is mostly useful when the delegate is a contract that implements its own rules for voting. These delegate-contracts
+ * can cast fractional votes according to the preferences of multiple entities delegating their voting power.
+ *
+ * Some example use cases include:
+ *
+ * * Voting from tokens that are held by a DeFi pool
+ * * Voting from an L2 with tokens held by a bridge
+ * * voting privately from a shielded pool using zero knowledge proofs.
+ *
+ * Based on ScopeLift's https://github.com/ScopeLift/flexible-voting/blob/e5de2efd1368387b840931f19f3c184c85842761/src/GovernorCountingFractional.sol
+ */
+abstract contract GovernorCountingFractional is Governor {
+    using Math for *;
+
```

Comment:
Following my previous comment, I'd suggest:

```suggestion
enum FractionalVoteType {
  Against
  For
  Abstain
  Fractional
}
```

I still believe it's handy to keep the Bravo vote type ordering.

Reply:
I'm not sure I understand. Current change keeps the Bravo ordering.

Reply:
The suggestion is just to add the enum definition here because I think it's specific to `GovernorCountingFractional`. I say I think it's handy that it uses the same order as Bravo although that's not strictly required.

Reply:
I expect most voters will not use params, and will instead relly on the canonical (bravo) methods. I think it is essential to not change the meaning of support for them.

Reply:
So I do. That's why this is **not suggesting to change the order**

Reply:
My comment was on "although that's not strictly required." ... IMO it is.

We all agree on not changing it!